src/libraries/System.Security.Cryptography.X509Certificates/tests/ExtensionsTests/AuthorityInformationAccessTests.cs

```diff
@@ -0,0 +1,267 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Formats.Asn1;
+using System.Linq;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+using Test.Cryptography;
+using Xunit;
+
+namespace System.Security.Cryptography.X509Certificates.Tests.ExtensionsTests
+{
+    public static class AuthorityInformationAccessTests
+    {
+        [Fact]
+        public static void DefaultConstructor()
+        {
+            X509AuthorityInformationAccessExtension aia = new();
+            Assert.NotNull(aia.Oid);
+            Assert.Equal("1.3.6.1.5.5.7.1.1", aia.Oid.Value);
+            Assert.Empty(aia.EnumerateCAIssuersUris());
+            Assert.Empty(aia.EnumerateOcspUris());
+        }
+
+        [Fact]
+        public static void NoProtocolFilter()
+        {
+            // While most of the callers of the various EnumerateUri methods will be interested in only the
+            // http(non-S) values, this test is just a quick sanity check that the built-in methods don't
+            // apply that filter internally.
+            using (X509Certificate2 cert = new X509Certificate2(TestFiles.MicrosoftRootCertFile))
+            {
+                X509AuthorityInformationAccessExtension aia =
+                    cert.Extensions.OfType<X509AuthorityInformationAccessExtension>().Single();
+
+                string[] caIssuersValues =
+                {
+                    "ldap:///CN=Microsoft%20Corporate%20Root%20Authority,CN=AIA,CN=Public%20Key%20Services,CN=Services,CN=Configuration,DC=Corp,DC=Microsoft,DC=Com?cACertificate?base?objectclass=certificationAuthority",
+                    "ldap://corp.microsoft.com/CN=Microsoft%20Corporate%20Root%20Authority,CN=AIA,CN=Public%20Key%20Services,CN=Services,CN=Configuration,DC=Corp,DC=Microsoft,DC=Com?cACertificate?base?objectclass=certificationAuthority",
+                    "http://www.microsoft.com/pki/mscorp/mscra1.crt",
+                };
+
+                const string CertificateAuthorityIssuers = "1.3.6.1.5.5.7.48.2";
+
+                Assert.Equal(caIssuersValues, aia.EnumerateCAIssuersUris());
+                Assert.Equal(caIssuersValues, aia.EnumerateUris(CertificateAuthorityIssuers));
+                Assert.Equal(caIssuersValues, aia.EnumerateUris(new Oid(CertificateAuthorityIssuers)));
+            }
+        }
+
+        [Fact]
+        public static void DecodeComplexValue()
```

Comment:
How about a test for testing decoding failures?

1. No sequence.
2. Sequence of something else.
3. Trailing data.